src/modules/Game.class.js

```diff
@@ -50,19 +158,97 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.#addNumber();
+    this.#addNumber();
+    this.#status = 'playing';
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.#applyInitialState();
+    this.#score = 0;
+    this.#status = 'idle';
+  }
+
+  #applyInitialState() {
+    this.#state = this.#initialState.map((row) => [...row]);
+  }
+  /**
+   * Checks if any move is possible
+   * @returns {boolean}
+   */
+  #isMovePossible() {
+    const gameCopy = new Game(this.getState(), true);
+
+    return (
+      gameCopy.moveLeft() ||
+      gameCopy.moveRight() ||
+      gameCopy.moveUp() ||
+      gameCopy.moveDown()
+    );
+  }
+  /**
+   * Returns if adding number to state was succesfull
+   * @returns {boolean}
+   */
+  #addNumber() {
+    const number = Math.random() > 0.1 ? 2 : 4;
+    const emptyFields = this.#findEmptyFields();
+
+    if (!emptyFields.length) {
+      return false;
+    }
 
-  // Add your own methods here
+    const randomIndex = Math.floor(Math.random() * emptyFields.length);
+    const { x, y } = emptyFields[randomIndex];
+
+    this.#state[y][x] = number;
+
+    return true;
+  }
+  /**
+   * Returns positions of empty fields in state
+   * @returns {({ x: number, y: number})[]}
+   */
+  #findEmptyFields() {
+    const result = [];
+
+    this.#state.forEach((row, y) => {
+      row.forEach((cell, x) => {
+        if (cell === 0) {
+          result.push({ x, y });
+        }
+      });
+    });
+
+    return result;
+  }
+
+  #reverse() {
+    this.#state.forEach((row) => row.reverse());
+  }
+
+  #rotate() {
+    this.#state = this.#state.map((row, i) => {
```

Comment:
The 'rotate' method can be optimized. Currently, it creates a new array for each row, which is not necessary. Instead, you could create a new matrix and fill it with the rotated values.